http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/auth/sam/PetClusterServiceAccountProvider.scala

```diff
@@ -1,34 +1,28 @@
 package org.broadinstitute.dsde.workbench.leonardo.auth.sam
 
-import com.typesafe.config.Config
+import cats.implicits._
+import cats.Monad
+import cats.mtl.ApplicativeAsk
+import org.broadinstitute.dsde.workbench.leonardo.dao.SamDAO
 import org.broadinstitute.dsde.workbench.leonardo.model.ServiceAccountProvider
-import org.broadinstitute.dsde.workbench.model.{UserInfo, WorkbenchEmail}
 import org.broadinstitute.dsde.workbench.model.google.GoogleProject
-
-import scala.concurrent._
-
-class PetClusterServiceAccountProvider(val config: Config) extends ServiceAccountProvider(config) with SamProvider {
-
-  override def getClusterServiceAccount(userInfo: UserInfo, googleProject: GoogleProject)(implicit executionContext: ExecutionContext): Future[Option[WorkbenchEmail]] = {
-    // Ask Sam for a pet service account for the given (user, project)
-    Future {
-      Option(blocking(samClient.getPetServiceAccount(userInfo, googleProject)))
-    }
+import org.broadinstitute.dsde.workbench.model.{TraceId, UserInfo, WorkbenchEmail}
+import org.http4s.headers.Authorization
+import org.http4s.{AuthScheme, Credentials}
+
+class PetClusterServiceAccountProvider[F[_]: Monad](sam: SamDAO[F]) extends ServiceAccountProvider[F] {
+  override def getClusterServiceAccount(userInfo: UserInfo, googleProject: GoogleProject)(implicit ev: ApplicativeAsk[F, TraceId]): F[Option[WorkbenchEmail]] = {
+    val authHeader = Authorization(Credentials.Token(AuthScheme.Bearer, userInfo.accessToken.token))
```

Comment:
Speculative question. We call some Sam APIs as the user, some as the pet service account, and others as the Leo service account. Should `HttpSamDAO` have knowledge about which user should call which API -- or should each method on `SamDAO` take an `Authorization` header, leaving it up to the caller? Asking as right now it seems like a mix of both, which is a little confusing (to me).

Reply:
yea, it was pretty confusing to me as well...but I'm not sure if we should differentiate them in `HttpSamDao` because the dao itself probably only cares that given a token, it calls sam, and sam can reject the call if token is invalid or token doesn't have enough permission.

Reply:
Agreed.. but in that case we should probably move `petTokenCache`, `getCachedPetAccessToken`, and `getAccessTokenUsingLeoPem` out of `HttpSamDAO`. Each method in Sam DAO should just take a token; callers are responsible for deciding which token to pass.

Reply:
I think https://broadworkbench.atlassian.net/browse/IA-1431 is a good opportunity for refactoring like this.